src/org/sosy_lab/cpachecker/cfa/mutation/strategy/RepeatedStrategy.java

```diff
@@ -0,0 +1,169 @@
+ *  This file is part of CPAchecker.
+ *
+ *  Copyright (C) 2007-2020  Dirk Beyer
+ *  All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.sosy_lab.cpachecker.cfa.mutation.strategy;
+
+import com.google.common.collect.ImmutableList;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.sosy_lab.common.configuration.Configuration;
+import org.sosy_lab.common.configuration.InvalidConfigurationException;
+import org.sosy_lab.common.log.LogManager;
+import org.sosy_lab.cpachecker.cfa.ParseResult;
+import org.sosy_lab.cpachecker.cfa.mutation.strategy.CompositeStrategy.CompositeStatistics;
+import org.sosy_lab.cpachecker.core.interfaces.Statistics;
+import org.sosy_lab.cpachecker.util.statistics.StatCounter;
+import org.sosy_lab.cpachecker.util.statistics.StatisticsWriter;
+
+public class RepeatedStrategy extends AbstractCFAMutationStrategy {
+  private OneRepeatStatistics thisCycle = new OneRepeatStatistics(1);
+  private final AbstractCFAMutationStrategy strategy;
+  private FullCycleStatistics stats;
+
+  private static class OneRepeatStatistics extends CompositeStatistics {
+    private final int cycle;
+
+    public OneRepeatStatistics(int pCycle) {
+      cycle = pCycle;
+    }
+  }
+
+  private static class FullCycleStatistics extends AbstractMutationStatistics {
+    private final StatCounter cycles = new StatCounter("cycles");
+    private final List<OneRepeatStatistics> cycleStats = new ArrayList<>();
+
+    @Override
+    public void printStatistics(PrintStream pOut, int indentLevel) {
+      StatisticsWriter w = StatisticsWriter.writingStatisticsTo(pOut).withLevel(indentLevel);
+      w.put(getName(), "")
+          .put(cycles)
+          .put(rounds)
+          .put(rollbacks)
+          .ifUpdatedAtLeastOnce(cycles)
+          .beginLevel();
+      for (OneRepeatStatistics c : cycleStats) {
+        w.put("Cycle " + c.cycle, "");
+        for (Statistics s : c.getStatistics()) {
+          ((AbstractMutationStatistics) s).printStatistics(pOut, indentLevel + 1);
+        }
+      }
+    }
+
+    @Override
+    public String getName() {
+      return "CycleStrategy";
+    }
+  }
+
+  public RepeatedStrategy(Configuration pConfig, LogManager pLogger)
+      throws InvalidConfigurationException {
+    super(pLogger);
+    strategy =
+        new CompositeStrategy(
```

Comment:
This Composite strategy is inside repeated strategy, which is inside composite. Do we need the complexity? Could we simplify the number of strategies?